src/form/remaining-chars-count/remaining-chars-count.directive.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.form.directive:pfRemainingCharsCount
+ *
+ * @description
+ *   Directive for showing a characters remaining count and triggering warning and error
+ *   behavior when passing specified thresholds.
+ *
+ * @param {string} ng-model The scope model variable which contains the intial text for the text field.  Required, but</br>
+ * can be an emptly string ("").
+ * @param {string} count-fld The id of the field to display the 'characters-remaining' count.
+ * @param {string} chars-max-limit Number representing the maximum number of characters to allow before dispatching a<br/>
```

Comment:
These parameters match what we put into the base PF ref. impl.
Isn't 'chars-max-limit' more descriptive?